src/components/MoneyRequestConfirmationList.js

```diff
@@ -88,244 +88,219 @@ const defaultProps = {
     ...withCurrentUserPersonalDetailsDefaultProps,
 };
 
-class MoneyRequestConfirmationList extends Component {
-    constructor(props) {
-        super(props);
-
-        const formattedParticipants = _.map(this.getParticipantsWithAmount(props.participants), (participant) => ({
-            ...participant,
-            selected: true,
-        }));
-
-        this.state = {
-            participants: formattedParticipants,
-            didConfirm: false,
-        };
-
-        this.toggleOption = this.toggleOption.bind(this);
-        this.confirm = this.confirm.bind(this);
-    }
-
+function MoneyRequestConfirmationList(props) {
     /**
-     * Get the confirmation button options
+     * Returns the participants with amount
+     * @param {Array} participants
      * @returns {Array}
      */
-    getSplitOrRequestOptions() {
-        const text = this.props.translate(this.props.hasMultipleParticipants ? 'iou.splitAmount' : 'iou.requestAmount', {
-            amount: CurrencyUtils.convertToDisplayString(this.props.iouAmount, this.props.iou.selectedCurrencyCode),
+    const getParticipantsWithAmount = useCallback(
+        (participantsList) => {
+            const iouAmount = IOUUtils.calculateAmount(participantsList.length, props.iouAmount);
+            return OptionsListUtils.getIOUConfirmationOptionsFromParticipants(participantsList, CurrencyUtils.convertToDisplayString(iouAmount, props.iou.selectedCurrencyCode));
+        },
+        [props.iouAmount, props.iou.selectedCurrencyCode],
+    );
+
+    const formattedParticipants = _.map(getParticipantsWithAmount(props.participants), (participant) => ({
+        ...participant,
+        selected: true,
+    }));
+
+    const [participants, setParticipants] = useState(formattedParticipants);
```

Comment:
`formattedParticipants` will be recalculated every rerender but won't update or change the value of `useState` and `formattedParticipants` isn't being used afterward. On the other hand, even if `getParticipantsWithAmount` is being recalculated because of any dependency change that won't modify the value of `useState`

I think would be better to move the `formattedParticipants` mapping into a function and run it into the `useState`

```
const getFormattedParticipants = () => {....}

const [participants, setParticipants] = useState(getFormattedParticipants);
```